### PR TITLE
Add Golang lib 

### LINKED ISSUE #7951: Add golang lib
### Kind of Issue

Other

### Description

I'm developing a new 3rd-party Golang lib to support simple icons in a different language

### Contributing

Yes

### PR DESCRIPTION
**Pull request by marcopollivier**
**Issue:** closes #7951 

### Description
Adding new lib to list of 3rd-party libs 

I created a new Golang lib to support projects that want to use the Simple Icons as a source of SVG icons. 

This lib can be found here https://github.com/marcopollivier/go-simple-icons

For a while, this lib returns a struct (similar to an object in Golang) but in future versions will support other formats too.

This lib was published as a community lib in the official Golang libs repository https://pkg.go.dev/github.com/marcopollivier/go-simple-icons

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -50,3 +50,6 @@ Icon
 Network Trash Folder
 Temporary Items
 .apdisk
+
+# IDE - Idea
+.idea
```
Comment (LitoMore) [inline .gitignore:55]: Do not append your personal settings.

You could use global ignore instead, see [**Configuring ignored files for all repositories on your computer**](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer).

File: README.md [hunk 2]
```diff
@@ -123,6 +123,7 @@ echo file_get_contents('path/to/package/icons/simpleicons.svg');
...
 | <img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/icons/flutter.svg#gh-light-mode-only" alt="Flutter" align=left width=24 height=24><img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/assets/readme/flutter-white.svg#gh-dark-mode-only" alt="Flutter" align=left width=24 height=24> [Flutter package](https://pub.dev/packages/simple_icons) | [@jlnrrg](https://jlnrrg.github.io/) |
+| <img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/icons/go.svg#gh-light-mode-only" alt="Go" align=left width=24 height=24><img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/assets/readme/go.svg#gh-dark-mode-only" alt="Go" align=left width=24 height=24> [Golang plugin WIP](https://github.com/marcopollivier/go-simple-icons/) | [@marcopollivier](https://github.com/marcopollivier) |
```
Comment (LitoMore) [inline README.md:126]: The filename should be `go-white.svg`

**Review by LitoMore — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -123,6 +123,7 @@ echo file_get_contents('path/to/package/icons/simpleicons.svg');
...
 | <img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/icons/flutter.svg#gh-light-mode-only" alt="Flutter" align=left width=24 height=24><img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/assets/readme/flutter-white.svg#gh-dark-mode-only" alt="Flutter" align=left width=24 height=24> [Flutter package](https://pub.dev/packages/simple_icons) | [@jlnrrg](https://jlnrrg.github.io/) |
+| <img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/icons/go.svg#gh-light-mode-only" alt="Go" align=left width=24 height=24><img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/assets/readme/go-white.svg#gh-dark-mode-only" alt="Go" align=left width=24 height=24> [Golang plugin WIP](https://github.com/marcopollivier/go-simple-icons/) | [@marcopollivier](https://github.com/marcopollivier) |
```
Comment (LitoMore) [inline README.md:126]: The `go-white.svg` needs to be added.

You need to copy-paste one from `icons/go.svg` and add `fill="white"` to `go-white.svg`

Comment (LitoMore) [reply README.md:126]:
```suggestion
| <img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/icons/go.svg#gh-light-mode-only" alt="Go" align=left width=24 height=24><img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/assets/readme/go-white.svg#gh-dark-mode-only" alt="Go" align=left width=24 height=24> [Golang plugin](https://github.com/marcopollivier/go-simple-icons/) | [@marcopollivier](https://github.com/marcopollivier) |
```

**Comment by LitoMore** [comment]
@marcopollivier What is your plan for bumping the version of Simple Icons? I see you just copy-paste the JSON file to your project and unmarshal it.

It would be better if you could add some automation work to sync our release to your project.

File: README.md [hunk 4]
```diff
@@ -123,6 +123,7 @@ echo file_get_contents('path/to/package/icons/simpleicons.svg');
...
 | <img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/icons/flutter.svg#gh-light-mode-only" alt="Flutter" align=left width=24 height=24><img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/assets/readme/flutter-white.svg#gh-dark-mode-only" alt="Flutter" align=left width=24 height=24> [Flutter package](https://pub.dev/packages/simple_icons) | [@jlnrrg](https://jlnrrg.github.io/) |
+| <img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/icons/go.svg#gh-light-mode-only" alt="Go" align=left width=24 height=24><img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/assets/readme/go-white.svg#gh-dark-mode-only" alt="Go" align=left width=24 height=24> [Golang plugin](https://github.com/marcopollivier/go-simple-icons/) | [@marcopollivier](https://github.com/marcopollivier) |
```
Comment (LitoMore) [inline README.md:126]:
```suggestion
| <img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/icons/go.svg#gh-light-mode-only" alt="Go" align=left width=24 height=24><img src="https://raw.githubusercontent.com/simple-icons/simple-icons/develop/assets/readme/go-white.svg#gh-dark-mode-only" alt="Go" align=left width=24 height=24> [Golang plugin](https://github.com/marcopollivier/go-simple-icons) | [@marcopollivier](https://github.com/marcopollivier) |
```

**Comment by marcopollivier** [comment]
> @marcopollivier What is your plan for bumping the version of Simple Icons? I see you just copy-paste the JSON file to your project and unmarshal it.
> 
> It would be better if you could add some automation work to sync our release to your project.

Good question. I'm thinking yet a better approach to do it. 
I thought of some options but I'm not sure about Them yet. 

- Create a submodule of the core lib and uses it as a reference. 
  - I'm not sure because I'm not a git-submodules big fan
  - I think if I do it I'll necessary import all of your codebase and it's not good
- if the submodule is the best option, ask you all if it's possible to isolate data in another repo (but it's not good too IMHO)
- Create a routine to copy and paste periodically from the main repo. Looks weird to me; or 

So, I really don't know yet. I'm looking another libs to understand their approach

**Comment by LitoMore** [comment]
@marcopollivier I created an issue for your project. Let's discuss this there: https://github.com/marcopollivier/go-simple-icons/issues/2.

**Comment by adamrusted** [comment]
Given no major update on this since Oct 22, I'm going to close this down. Feel free to open once issues reported by @LitoMore are resolved, @marcopollivier!

**Pull request closed without merging**